sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -75,9 +85,43 @@ public DefaultService(ServiceSpec serviceSpec, Collection<Plan> plans) {
         register(serviceSpec, this.plans);
     }
 
+    private void lock() {
+        String rootPath = CuratorUtils.toServiceRootPath(serviceSpec.getName());
+        String lockPath = CuratorUtils.join(rootPath, "LOCK_PATH");
+        curatorMutex = new InterProcessMutex(curatorClient, lockPath);
+
+        LOGGER.info("Acquiring ZK lock...");
```

Comment:
Could include lockPath here and in the error messages below, to help debuggability just in case.